assets/app/view/game/buy_trains.rb

```diff
@@ -70,12 +70,12 @@ def render
         step = @game.round.active_step
         @corporation = step.current_entity
         if @selected_company&.owner == @corporation
-          @ability = @game.abilities(@selected_company, :train_discount, time: 'train')
+          @ability = @game.abilities(@selected_company, :train_discount, time: 'buying_train')
         end
 
         @depot = @game.depot
 
-        available = step.buyable_trains(@corporation).group_by(&:owner)
+        available = step.buyable_trains(@corporation).group_by(&:owner) || []
```

Comment:
this can never be nil why did you change this

Reply:
don't remember, overlooked it when I was reviewing this myself last night, will delete